client/signup/config/test/index.js

```diff
@@ -32,3 +36,64 @@ describe( 'index', () => {
 		}
 	} );
 } );
+
+describe( 'Signup flows configuration validation', () => {
+	let FlowController, ProgressStore, DependencyStore, Flows, flowControllerInstance;
+
+	useFakeDom();
+
+	useMockery( ( mockery ) => {
+		mockery.registerMock( 'lib/abtest', {
+			abtest: identity,
+			getABTestVariation: identity,
+		} );
+		mockery.registerMock( 'lib/signup/step-actions', {} );
+	} );
+
+	before( () => {
+		ProgressStore = require( 'lib/signup/progress-store' );
+		DependencyStore = require( 'lib/signup/dependency-store' );
+		FlowController = require( 'lib/signup/flow-controller' );
+		Flows = require( 'signup/config/flows' );
+		ProgressStore.reset();
+		const store = createStore( reducer );
+		DependencyStore.setReduxStore( store );
+
+
+		// temporarily instantiate the FlowController with a valid flow
+		flowControllerInstance = FlowController( {
+			flowName: 'account'
+		} );
+	} );
+
+	describe( 'Signup flows configuration validation', () => {
+		Object.keys( flowsData ).forEach( ( flowName ) => {
+			describe( `Verifying flow "${ flowName }"`, () => {
+				afterEach( function() {
+					flowControllerInstance.reset();
+					DependencyStore.reset();
+					ProgressStore.reset();
+				} );
+
+				it( 'Valid configuration', () => {
+					try {
+						flowControllerInstance = FlowController( {
+							flowName: flowName,
+							onComplete: () => {
+								console.log( 'Complete flow' );
```

Comment:
I was a little thrown off by this line. Removing the `log` statement didn't change the test outcome for me (I think).

What I then did try, due to seeing the name `onComplete`, was the following:

```diff
diff --git a/client/signup/config/test/index.js b/client/signup/config/test/index.js
index ee4bb5ec2..0caa98d37 100644
--- a/client/signup/config/test/index.js
+++ b/client/signup/config/test/index.js
@@ -75,12 +75,12 @@ describe( 'Signup flows configuration validation', () => {
 					ProgressStore.reset();
 				} );
 
-				it( 'Valid configuration', () => {
+				it( 'Valid configuration', ( done ) => {
 					try {
 						flowControllerInstance = FlowController( {
 							flowName: flowName,
 							onComplete: () => {
-								console.log( 'Complete flow' );
+								done();
 							}
 						} );
 					} catch ( Exception ) {
```

but this results in all tests timing out. As I'm mostly unfamiliar with Signup, my questions are:

- Do we not expect the controller instance to _complete_?
- If we don't, do we need to define a `onComplete` function? If so, can it be `noop`?
- If we don't really reach `onComplete`, then we're really mostly testing by making sure no errors are thrown in that same stack. Layman's question: is that enough?